brd-ios/breadwallet/src/Redesign/Scenes/Swap/Details/SwapDetailsPresenter.swift

```diff
@@ -16,89 +16,169 @@ final class SwapDetailsPresenter: NSObject, Presenter, SwapDetailsActionResponse
+                                                       logo: detail.source.paymentInstrument.displayImage,
+                                                       cardNumber: .text(detail.source.paymentInstrument.displayName),
+                                                       expiration: .text(CardDetailsFormatter.formatExpirationDate(month: detail.source.paymentInstrument.expiryMonth,
+                                                                                                                   year: detail.source.paymentInstrument.expiryYear)),
+                                                       cvvTitle: nil, cvv: nil))
             ],
             Models.Section.fromCurrency: [
                 AssetViewModel(icon: fromImage,
-                               title: "From \(item.source.currency)",
+                               title: "From \(detail.source.currency)",
                                topRightText: "\(formattedCurrencyAmountString) / $\(formattedUsdAmountString) \(currencyCode)")
             ],
             Models.Section.image: [
                 ImageViewModel.imageName("arrowDown")
             ],
             Models.Section.toCurrency: [
-                AssetViewModel(icon: toImage,
-                               title: "To \(item.destination.currency)",
-                               topRightText: "\(formattedCurrencyAmountDestination) / $\(formattedUsdAmountDestination) \(currencyCode)")
+                toCurrencyAssetViewModel
             ],
             Models.Section.timestamp: [
                 TransactionViewModel(title: "Timestamp",
                                      description: "\(dateString)")
             ],
             Models.Section.transactionFrom: [
-                OrderViewModel(title: "\(item.source.currency) Transaction ID",
-                               value: "\(String(describing: item.source.transactionId))",
-                               imageName: "copy")
+                OrderViewModel(title: "\(detail.source.currency) Transaction ID",
+                               value: SwapDetailsPresenter.generateAttributedOrderValue(with: transactionFromValue),
+                               showsFullValue: true)
             ],
             Models.Section.transactionTo: [
-                TransactionViewModel(title: "\(item.destination.currency) Transaction ID",
-                                     description: "\(item.status.rawValue.localizedCapitalized)")
+                TransactionViewModel(title: "\(detail.destination.currency) Transaction ID",
+                                     description: "\(detail.status.rawValue.localizedCapitalized)")
             ]
         ]
         
         viewController?.displayData(responseDisplay: .init(sections: sections, sectionRows: sectionRows))
     }
 
     // MARK: - Additional Helpers
-
+    
+    private static func generateAttributedOrderValue(with value: String) -> NSAttributedString {
```

Comment:
what exactly is this? 🤔

Reply:
![image](https://user-images.githubusercontent.com/99484553/187368529-9e01c15d-8ef4-46a4-a710-82ca4e11cc0e.png)

Creates those strings with the "copy" icon right next to them.